django/contrib/gis/geos/prototypes/io.py

```diff
@@ -178,19 +179,28 @@ def read(self, wkb):
             raise TypeError
 
 
+def default_trim_value():
+    """
+    GEOS changed the default value in 3.12.0
+    can be replaced by a static True when 3.12.0 becomes the minimum supported version
+    """
+    return geos_version_tuple() >= (3, 12)
+
+
+DEFAULT_TRIM_VALUE = SimpleLazyObject(default_trim_value)
+
+
 # ### WKB/WKT Writer Classes ###
 class WKTWriter(IOBase):
     _constructor = wkt_writer_create
     ptr_type = WKT_WRITE_PTR
     destructor = wkt_writer_destroy
-
-    _trim = False
     _precision = None
 
     def __init__(self, dim=2, trim=False, precision=None):
         super().__init__()
-        if bool(trim) != self._trim:
-            self.trim = trim
+        self._trim = bool(DEFAULT_TRIM_VALUE)
+        self.trim = trim
```

Comment:
I moved this to a separate PR, check out #17063.

Reply:
merged

Reply:
we still need to set the default trim value in an object instance variable so that we can detect the `trim` parameter value is different from what is stored in the underlying GEOS object, so that the function `wkt_writer_set_trim` is called appropriately depending on GEOS version, i mean:
- if geos < 3.12 and trim parameter is False, do not call wkt_writer_set_trim
- if geos >= 3.12 and trim parameter is True, do not call wkt_writer_set_trim